files/brews/nginx.rb

```diff
@@ -1,67 +1,206 @@
+      --with-http_ssl_module
+      --with-pcre
+      --sbin-path=#{bin}/nginx
+      --with-cc-opt=#{cc_opt}
+      --with-ld-opt=#{ld_opt}
+      --conf-path=#{etc}/nginx/nginx.conf
+      --pid-path=#{var}/run/nginx.pid
+      --lock-path=#{var}/run/nginx.lock
+      --http-client-body-temp-path=#{var}/run/nginx/client_body_temp
+      --http-proxy-temp-path=#{var}/run/nginx/proxy_temp
+      --http-fastcgi-temp-path=#{var}/run/nginx/fastcgi_temp
+      --http-uwsgi-temp-path=#{var}/run/nginx/uwsgi_temp
+      --http-scgi-temp-path=#{var}/run/nginx/scgi_temp
+      --http-log-path=#{var}/log/nginx/access.log
+      --error-log-path=#{var}/log/nginx/error.log
+      --with-http_gzip_static_module
+      --with-http_v2_module
+    ]
+
+    if build.with? "passenger"
+      nginx_ext = `#{Formula["passenger"].opt_bin}/passenger-config --nginx-addon-dir`.chomp
+      args << "--add-module=#{nginx_ext}"
+    end
+
+    args << "--with-http_dav_module" if build.with? "webdav"
+    args << "--with-debug" if build.with? "debug"
+    args << "--with-http_gunzip_module" if build.with? "gunzip"
+
+    if build.head?
+      system "./auto/configure", *args
+    else
+      system "./configure", *args
+    end
+
+    system "make", "install"
+    if build.head?
+      man8.install "docs/man/nginx.8"
+    else
+      man8.install "man/nginx.8"
+    end
+  end
+
+  def post_install
+    (etc/"nginx/servers").mkpath
+    (var/"run/nginx").mkpath
+
+    # nginx's docroot is #{prefix}/html, this isn't useful, so we symlink it
+    # to #{HOMEBREW_PREFIX}/var/www. The reason we symlink instead of patching
+    # is so the user can redirect it easily to something else if they choose.
+    html = prefix/"html"
+    dst = var/"www"
+
+    if dst.exist?
+      html.rmtree
+      dst.mkpath
+    else
+      dst.dirname.mkpath
+      html.rename(dst)
+    end
+
+    prefix.install_symlink dst => "html"
+
+    # for most of this formula's life the binary has been placed in sbin
+    # and Homebrew used to suggest the user copy the plist for nginx to their
+    # ~/Library/LaunchAgents directory. So we need to have a symlink there
+    # for such cases
+    if rack.subdirs.any? { |d| d.join("sbin").directory? }
+      sbin.install_symlink bin/"nginx"
+    end
+  end
+
+  def passenger_caveats; <<-EOS.undent
+    To activate Phusion Passenger, add this to #{etc}/nginx/nginx.conf, inside the 'http' context:
+      passenger_root #{Formula["passenger"].opt_libexec}/src/ruby_supportlib/phusion_passenger/locations.ini;
+      passenger_ruby /usr/bin/ruby;
+    EOS
+  end
+
+  def caveats
+    s = <<-EOS.undent
+    Docroot is: #{var}/www
+
+    The default port has been set in #{etc}/nginx/nginx.conf to 8080 so that
+    nginx can run without sudo.
+
+    nginx will load all files in #{etc}/nginx/servers/.
+    EOS
+    s << "\n" << passenger_caveats if build.with? "passenger"
+    s
+  end
+
+  plist_options :manual => "nginx"
+
+  def plist; <<-EOS.undent
```

Comment:
This `plist` entry isn't needed since we are already managing it via [`templates/dev.nginx.plist.erb`](https://github.com/boxen/puppet-nginx/blob/master/templates/dev.nginx.plist.erb)